src/chains/ethereum/ethereum/tests/api/eth/feeHistory.test.ts

```diff
@@ -0,0 +1,539 @@
+            const blockCount = "0xa";
+            const newestBlock = "0x2";
+            const feeHistory = await provider.send("eth_feeHistory", [
+              blockCount,
+              newestBlock,
+              []
+            ]);
+
+            assert.equal(feeHistory.oldestBlock, genesisBlock);
+            assert.equal(feeHistory.baseFeePerGas.length, 4); // blocks found + 1
+            assert.equal(feeHistory.gasUsedRatio.length, 3);
+          });
+          it("newestBlock = oldestBlock when blockCount === 1", async () => {
+            const blockCount = "0x1";
+            const newestBlock = "0x2";
+            const feeHistory = await provider.send("eth_feeHistory", [
+              blockCount,
+              newestBlock,
+              []
+            ]);
+
+            assert.equal(feeHistory.oldestBlock, newestBlock);
+          });
+          it("oldestBlock = newestBlock - blockCount +1", async () => {
+            const blockCount = "0x1";
+            const newestBlock = "0x2";
+            const feeHistory = await provider.send("eth_feeHistory", [
+              blockCount,
+              newestBlock,
+              []
+            ]);
+
+            assert.equal(feeHistory.oldestBlock, newestBlock);
+          });
+        });
+      });
+      describe("gasUsedRatio", () => {
+        it("returns 0 gasUsed for empty blocks", async () => {
+          const blockCount = "0x1";
+          const newestBlock = "0x2";
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          assert.equal(feeHistory.gasUsedRatio[0], 0);
+        });
+        it("returns gasUsed / maxGas", async () => {
+          await sendTransaction({ provider, to, from });
+
+          const block = await provider.send("eth_getBlockByNumber", ["latest"]);
+
+          const blockCount = "0x1";
+          const newestBlock = block.number;
+
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          assert.equal(feeHistory.gasUsedRatio.length, 1);
+          assert.equal(
+            feeHistory.gasUsedRatio[0],
+            Number(block.gasUsed) / Number(block.gasLimit)
+          );
+        });
+        it("returns one entry for each block", async () => {
+          await mineNBlocks({ provider, blocks: 1 });
+          await sendTransaction({ provider, to, from });
+          await mineNBlocks({ provider, blocks: 1 });
+          const block = await provider.send("eth_getBlockByNumber", ["latest"]);
+
+          const blockCount = "0x2";
+          const newestBlock = block.number;
+
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          assert.equal(feeHistory.gasUsedRatio.length, 2);
```

Comment:
Could use `deepStrictEquals` here to avoid checking each of the individual properties